src/components/UrqlQuery.re

```diff
@@ -1,70 +1,57 @@
-[@bs.module "urql"] external queryComponent: ReasonReact.reactClass = "Query";
-
+[@bs.deriving abstract]
 type queryRenderPropsJs('a) = {
-  .
-  "fetching": bool,
-  "data": Js.Nullable.t('a),
-  "error": Js.Nullable.t(UrqlCombinedError.t),
-  "executeQuery":
-    option(Js.Json.t) => Js.Promise.t(UrqlTypes.operationResult),
+  fetching: bool,
+  [@bs.optional] data: 'a,
+  [@bs.optional] error: UrqlCombinedError.t,
+  executeQuery: option(Js.Json.t) => Js.Promise.t(UrqlTypes.operationResult)
 };
 
 type queryRenderProps('a) = {
   fetching: bool,
   data: option('a),
   error: option(UrqlCombinedError.t),
   executeQuery: option(Js.Json.t) => Js.Promise.t(UrqlTypes.operationResult),
-  response: UrqlTypes.response('a),
+  response: UrqlTypes.response('a)
 };
 
+module QueryJs = {
+  [@bs.module "urql"] [@react.component]
+  external make:
+    (~query: string, ~variables: Js.Json.t, ~requestPolicy: string, ~children: queryRenderPropsJs('a) => React.element) =>
+    React.element =
+    "Query";
+}
+
 let urqlDataToRecord = (result: queryRenderPropsJs('a)) => {
-  let data = result##data |> Js.Nullable.toOption;
-  let error = result##error |> Js.Nullable.toOption;
+  let data = result->dataGet;
```

Comment:
With the `[@bs.optional]` annotation do these get wrapped in `option` automatically? Based on the pattern match below that appears to be what's happening, which is really nice!

Reply:
It does! And also play nice with JavaScript object interop. So we have the benefits of type safety and type checking, while being easy to interop with JS obj.
https://bucklescript.github.io/docs/en/object#record-mode